python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -505,6 +505,28 @@ abstract class FunctionValue extends CallableValue {
     predicate isOverriddenMethod() {
         exists(Value f | f.overrides(this))
     }
+    
+    /** Whether `name` is a legal argument name for this function */
+    bindingset[name]
+    predicate isLegalArgumentName(string name) {
+        this.getScope().getAnArg().asName().getId() = name
+        or
+        this.getScope().getAKeywordOnlyArg().getId() = name
+        or
+        this.getScope().hasKwArg()
+    }
+    
+    /** Whether this is a "normal" method, that is, it is exists as a class attribute 
+     *  which is not wrapped and not the __new__ method. */
+    predicate isNormalMethod() {
+        exists(ClassValue cls, string name |
+            cls.declaredAttribute(name) = this and
+            name != "__new__" and
+            exists(Expr expr, AstNode origin | expr.pointsTo(this, origin) |
+              not origin instanceof Lambda
```

Comment:
Why are lambdas excluded?

Reply:
My best guess is that these give false positives otherwise. A test would be to try disabling this conjunct and seeing whether it results in any differences in query output.

Reply:
However: with the original version, they also seem to be failing on my machine. So I cannot say what's going on. Should I push the modified version for testing on Jenkins to see what it says?

Reply:
I see you made https://github.com/Semmle/ql/pull/3003, and that all tests passed without the lambda. But I guess that is expected, since `isNormalMethod` is only used by the newly added predicates in `CallArgs.qll` which aren't used in any queries yet (at least not in this PR), so there is no tests that _could_ fail because of this.